internal/sandbox/init.go

```diff
@@ -0,0 +1,130 @@
+package sandbox
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/ossf/package-analysis/internal/log"
+)
+
+const (
+	ipBin           = "/usr/sbin/ip"
+	iptablesLoadBin = "/usr/sbin/iptables-restore"
+	iptablesRules   = "/usr/local/etc/iptables.rules"
+	dummyInterface  = "cnidummy0"
+
+	// bridgeInterface is the name of the podman bridge defined in
+	// tools/network/podman-analysis.conflist. This bridge is used by the
+	// sandbox during analysis to separate the sandbox traffic from the host.
+	bridgeInterface = "cni-analysis"
+)
+
+const (
+	// NetworkInterface is the name of a network interface that has access to
+	// the sandbox network traffic.
+	NetworkInterface = bridgeInterface
+)
+
+func loadIptablesRules() error {
+	log.Debug("Loading iptable rules")
+
+	// Open the iptables-restore configuration
+	f, err := os.Open(iptablesRules)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	logOut := log.Writer(log.InfoLevel)
+	defer logOut.Close()
+	logErr := log.Writer(log.WarnLevel)
+	defer logErr.Close()
+
+	cmd := exec.Command(iptablesLoadBin)
+	cmd.Stdout = logOut
+	cmd.Stderr = logErr
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+	defer stdin.Close()
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+	// Send the iptables rules to the command via stdin
+	if _, err := io.Copy(stdin, f); err != nil {
+		return err
+	}
+	stdin.Close()
+	return cmd.Wait()
+}
+
+// createBridgeNetwork ensures that NetworkInterface and the bridge network
+// exists prior to the sandbox.
+//
+// podman would create this bridge interface anyway, but doing it early allows
+// a packet capture to be started on the interface prior to the sandbox
+// starting.
+func createBridgeNetwork() error {
+	log.Debug("Creating bridge network")
+
+	// Create the bridge
+	cmd := exec.Command(ipBin, "link", "add", "name", bridgeInterface, "type", "bridge")
```

Comment:
Are these things we can set up to persist in the Dockerfile/container instead?

Reply:
The only way we could automate this is to have an "init.sh" script in the image that serves as the entrypoint.

There is no standard approach to doing this without systemd or some other init process.

Happy to do that if it makes more sense.

Reply:
Ack, thanks! This seems fine to me if the only way to get this to work is via an entrypoint.